src/cpp/session/modules/NotebookData.R

```diff
@@ -52,7 +52,7 @@
 
     max.print <- if (is.null(options$max.print)) getOption("max.print", 1000) else options$max.print
 
-    x <- head(x, max.print)
+    x <- as.data.frame(head(x, max.print))
```

Comment:
Should we use `.rs.toDataFrame` instead? `as.data.frame` is an S3 method so at a minimum here we should probably evaluate in the global environment so that any package-specific code for coercion gets executed.